### PR TITLE
Handle 500 errors that include an error message.

### LINKED ISSUE #360: Not all errors passed through to error message
I'm seeing a lot of errors passed back to my callbacks that include a null message. Eg:

```
{ [Error: null] code: 500 }
```

However, when examining the body of the actual HTTP response, I see that the message is actually populated:

Actual response object:
```
 {
   "statusCode": 500,
   "body": {
     "error": {
       "errors": [
         {
           "domain": "global",
           "reason": "backendError",
           "message": "Backend Error"
         }
       ],
       "code": 500,
       "message": "Backend Error"
     }
   }
 }
```

### PR DESCRIPTION
**Pull request by bradvogel**
Fixes #360

### CODE REVIEW

File: test/test.transporters.js [hunk 1]
```diff
@@ -124,4 +124,22 @@ it('should return errors within response body as instances of Error', function(d
       done();
     });
   });
+
+  it('should handle 5xx responses that include errors', function(done) {
+    var google = require('../lib/googleapis');
+    var scope = nock('https://www.googleapis.com')
+      .post('/urlshortener/v1/url')
+      .reply(500, { error: {message: 'There was an error!'}});
```
Comment (ryanseys) [inline test/test.transporters.js]: Should also test with your original object you were having issues with:

``` js
 {
   "statusCode": 500,
   "body": {
     "error": {
       "errors": [
         {
           "domain": "global",
           "reason": "backendError",
           "message": "Backend Error"
         }
       ],
       "code": 500,
       "message": "Backend Error"
     }
   }
 }
```

File: lib/transporters.js [hunk 2]
```diff
@@ -73,29 +73,28 @@ DefaultTransporter.prototype.wrapCallback_ = function(opt_callback) {
...
-      // handle single request errors
+    // Consider the response an error if it includes the 'error' property or it has a
+    // 500 status code.
+    if ((body && body.error) || res.statusCode >= 500) {
 
-      if (typeof body.error === 'string') {
+      if (!body.error){
+        err = new Error(body);
+        err.code = res.statusCode;
+        err.type = body.error;
```
Comment (ryanseys) [inline lib/transporters.js]: This is null or undefined.

File: lib/transporters.js [hunk 3]
```diff
@@ -73,29 +73,28 @@ DefaultTransporter.prototype.wrapCallback_ = function(opt_callback) {
       body = JSON.parse(body);
     } catch (err) { /* no op */ }
 
-    if (body && body.error) {
-      // handle single request errors
+    // Consider the response an error if it includes the 'error' property or it has a
+    // 500 status code.
+    if ((body && body.error) || res.statusCode >= 500) {
```
Comment (ryanseys) [inline lib/transporters.js]: I don't really like this nested if block. I believe we could achieve the same thing with an if-else block

File: test/test.transporters.js (hunk 1 above)
Comment (bradvogel) [inline test/test.transporters.js]: Added below.

Comment (ryanseys) [inline test/test.transporters.js]: Isn't the whole body the entire object I put above? Not just the body component of that object? I might be mistaken.

**Comment by bradvogel** [comment]
Updated.

File: test/test.transporters.js (hunk 1 above)
Comment (bradvogel) [inline test/test.transporters.js]: The JSON I pasted (which you have above) is the complete JSON representation of the response object. So only the 'body' part is the JSON returned in the response body.

Comment (ryanseys) [inline test/test.transporters.js]: Ah my bad.

**Comment by bradvogel** [comment]
Thanks! When are you planning to push another version?

**Comment by ryanseys** [comment]
Released.

**Pull request merged**